### PR TITLE
【BIT】转换能力补齐：paddle.nn.functional.[margin_ranking_loss class_center_sample gelu]

### PR DESCRIPTION
**Pull request by cangtianhuang**
### 1. [paddle.nn.functional.margin_ranking_loss](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api/paddle/nn/functional/margin_ranking_loss_cn.html)

查询 [torch.nn.functional.margin_ranking_loss 与 paddle.nn.functional.margin_ranking_loss 对照表](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/guides/model_convert/convert_from_pytorch/api_difference/functional/torch.nn.functional.margin_ranking_loss.html) ，发现是 `torch 参数更多`，测试后发现 `label` 参数形状需要与 `input` 一致，因此在直接映射的配置中进行广播转写。

### 2. [paddle.nn.functional.class_center_sample](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api/paddle/nn/functional/class_center_sample_cn.html)

查询对照表与资料，发现并没有对应的 torch api ，因此直接为其注册 ClassCenterSampleRule 类，编写转换代码。

由于 `label` 参数介绍为：
> label (Tensor) - 1-D Tensor，数据类型为 int32 或者 int64，每个元素的取值范围在 [0, num_classes)。

因此为其在 config_analyzer 中增加了分支处理，限制了标签的生成范围（即使之前 paddleonly 能够通过测试）；然后测试 accuracy 发现部分配置未通过，原因是该 api 会随机采样负类别中心补够 num_samples，具有随机性，因此将配置均移入 paddleonly 中（但是支持 paddle2torch 转换）

### 3. [paddle.nn.functional.gelu](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api/paddle/nn/functional/gelu_cn.html)

查询 [torch.nn.functional.gelu 与 paddle.nn.functional.gelu 对照表](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/guides/model_convert/convert_from_pytorch/api_difference/functional/torch.nn.functional.gelu.html) ，发现是 `参数用法不一致`，但可通过配置将 `approximate` 参数转写，因此为其撰写直接映射。

## 测试结果

paddle.nn.functional.class_center_sample 配置全部移入 paddleonly_1 中；paddle.nn.functional.gelu 有一条配置 OOM，暂时移入 accuracy_error_1 中；其余配置均通过测试，移入 support2torch_1 中

image

### CODE REVIEW

File: tester/api_config/api_config_accuracy_error_1.txt [hunk 1]
```diff
@@ -8,3 +8,4 @@ paddle.nn.functional.batch_norm(Tensor([4, 6, 12, 24],"float64"), Tensor([6],"fl
 paddle.nn.functional.batch_norm(Tensor([8, 8, 16, 16],"float32"), Tensor([8],"float32"), Tensor([8],"float32"), Tensor([8],"float32"), Tensor([8],"float32"), training=True, momentum=0.9, epsilon=1e-05, data_format="NCHW", use_global_stats=True, )
 paddle.nn.functional.batch_norm(Tensor([8, 8, 16, 16],"float64"), Tensor([16],"float64"), Tensor([16],"float64"), Tensor([16],"float64"), Tensor([16],"float64"), training=False, momentum=0.9, epsilon=1e-05, data_format="NHWC", use_global_stats=False, )
 paddle.nn.functional.batch_norm(Tensor([8, 8, 16, 16],"float64"), Tensor([16],"float64"), Tensor([16],"float64"), Tensor([16],"float64"), Tensor([16],"float64"), training=True, momentum=0.9, epsilon=1e-05, data_format="NHWC", use_global_stats=True, )
+paddle.nn.functional.gelu(Tensor([128, 96, 96, 768],"float32"), False, None, )
```
Comment (cszdrg) [inline tester/api_config/api_config_accuracy_error_1.txt:11]: 这在report里也放一份

Comment (cangtianhuang) [reply tester/api_config/api_config_accuracy_error_1.txt:11]: OOM 也需要放吗？我不清楚其他人的 OOM 是否是放在 accuracy中？

**Comment by cszdrg** [comment]
我看你的截图里不是有一个error吗？为什么没放进去

**Comment by cangtianhuang** [comment]
> paddle.nn.functional.class_center_sample

[paddle.nn.functional.class_center_sample](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api/paddle/nn/functional/class_center_sample_cn.html) api 具有随机性，后续将所有配置放入 paddleonly_1 中了

**Review by cszdrg — APPROVED** [review]
